### DISCUSSION TITLE
关于使用FCP-vasp-ase进行BOMD的疑问

### DISCUSSION THREAD

**Discussion by yhzhao1989** [hellozhaoming/FCP-vasp-ase#4] [Q&A]
夏老师，您好。最近在拜读您的文章和代码。您的FCP能够很好快速地去做GC-DFT的计算，并且还可以实现相应的BOMD。但是我对于MD这一块有一部分的不解，我目前使用的是您version2的版本。在跑MD时参考您version1给出跑MD的脚本，结果是能运行。但是我不是很确定自己修改后的正确性。主要疑惑的地方如下：
不同于一般的AIMD直接由vasp全程实现。在考虑电势的BOMD实现，其实是结合了vasp和ase的MD模块实现的。但ase获取原子受力的原理是否和vasp实现MD时是一样的呢，或者两者是等价的？我并没有在ASE的手册中看到相关的描述。
附件中是我自己修改了的脚本，希望您能帮我答疑解惑。
[md.zip](https://github.com/user-attachments/files/15510907/md.zip)

**Comment by hellozhaoming** [comment]
我个人的感觉是永远不要完全信任任何一个软件，我无法确保自己写的代码能够适用于任何体系或者不存在任何bug。
目前ase能够读取vasp的受力做结构优化，但是优化的算法不一样，得到的能量极小值点可能不一样。你可以做简单的测试，用ase优化的结构再给vasp做结构优化，看看能否找到能量更低的结构，或者用vasp优化得到的结构再去拿给ase做结构优化，看看能够找到能量更低的结构。
ase做MD的模块我测试过，一般问题不大，但是对于有原子固定的md，我测试会有bug，它的温度无法收敛到设定的值。原因是温度统计的是没有固定的原子的自由度的平均动能，但ase里md.py把所有原子的自由度都考虑了。
我尝试过修改ase的源码并请求合并到主分支，但是因为我没有给出具体测试案例被拒了，后来太忙了就忘记了。

你如果md的时候要固定原子也需要md.py做相应的修改。

**Reply by yhzhao1989** [reply]
好的，我明白了。那还是得多测试测试。感谢夏老师的认真回复和提醒。
